dev-tools/check_file_permissions.sh

```diff
@@ -6,7 +6,7 @@ files_with_wrong_permissions=$(
         ':!*.sh' \
         ':!php-cs-fixer' \
         ':!dev-tools/*.php' \
-    | grep -P "^100755 " \
```

Comment:
what was wrong with perl-mode?

Reply:
It's described in the 8b9a1c41.

<img width="782" alt="image" src="https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/600668/d1611760-93bd-417e-9f81-393e647fb6ee">

Reply:
![Screenshot 2023-08-11 at 18 21 44](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/2716794/7a23d98a-4426-4ba5-8493-0a4f75cfc708)

Reply:
```
ker@d ~/github/FriendsOfPHP/PHP-CS-Fixer λ sw_vers
ProductName:	macOS
ProductVersion:	12.6.2
BuildVersion:	21G320
ker@d ~/github/FriendsOfPHP/PHP-CS-Fixer λ system_profiler SPSoftwareDataType
Software:

    System Software Overview:

      System Version: macOS 12.6.2 (21G320)
      Kernel Version: Darwin 21.6.0
      ...

ker@d ~/github/FriendsOfPHP/PHP-CS-Fixer λ 
ker@d ~/github/FriendsOfPHP/PHP-CS-Fixer λ 
ker@d ~/github/FriendsOfPHP/PHP-CS-Fixer λ grep --version
grep (GNU grep) 3.11
Packaged by Homebrew
Copyright (C) 2023 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Mike Haertel and others; see
<https://git.savannah.gnu.org/cgit/grep.git/tree/AUTHORS>.

grep -P uses PCRE2 10.42 2022-12-11
ker@d ~/github/FriendsOfPHP/PHP-CS-Fixer λ 
ker@d ~/github/FriendsOfPHP/PHP-CS-Fixer λ 
ker@d ~/github/FriendsOfPHP/PHP-CS-Fixer λ grep --help | grep "perl"         
  -P, --perl-regexp         PATTERNS are Perl regular expressions
ker@d ~/github/FriendsOfPHP/PHP-CS-Fixer λ 
```

Reply:
overall, I'm ok for the change, but I prefer the perl regex than this ugly extended posix format as a rule

Reply:
I am not regex expert and probably don't know how exactly extended regex compares to Perl one, but in this particular case it does not make any difference in practice. I have M1 Mac, and I believe I use built-in (non-GNU) `grep` so IMHO it's better to ship script that works out of the box, without the need to install Brew package.

Reply:
FYI:

```shell
$ sw_vers
ProductName:		macOS
ProductVersion:		13.4.1
ProductVersionExtra:	(c)
BuildVersion:		22F770820d

$ system_profiler SPSoftwareDataType
Software:

    System Software Overview:

      System Version: macOS 13.4.1 (c) (22F770820d)
      Kernel Version: Darwin 22.5.0

$ grep --version
grep (BSD grep, GNU compatible) 2.6.0-FreeBSD
```